src/integration/rustAnalyzer.ts

```diff
@@ -1,23 +1,33 @@
 import { Manifest } from '../manifest'
+import { Integration } from '.'
+import * as vscode from 'vscode'
 
-export async function loadRustAnalyzerConfigOverrides(manifest: Manifest): Promise<void> {
-    await manifest.overrideWorkspaceCommandConfig('rust-analyzer', 'server.path', 'rust-analyzer', '/usr/lib/sdk/rust-stable/bin/')
+export class RustAnalyzer implements Integration {
+    isApplicable(manifest: Manifest): boolean {
+        const isUsingRustSdk = manifest.sdkPath()?.includes('rust') || false
+        const isRustAnalyzerExtEnabled = vscode.extensions.getExtension('matklad.rust-analyzer') !== undefined
```

Comment:
As all the integrations would check for an extension being available, I would move that to a provided function of Integration interface and only do the check if `isApplicable`  is true. That would avoid us always doing the extensions checks which could maybe slow things. 

Alternatively, we only need to provide two things: the supported sdk extension & the vscode extension id. So isApplicable could be a generic function that makes use of two the interface properties.

Reply:
There might be more reason though to determine if an integration is applicable like #43

Reply:
> There might be more reason though to determine if an integration is applicable like #43

That could be also automatically checked once that lands and would simplify the whole process of adding new integrations